addons/hr_holidays_contract/tests/test_accrual_allocations_with_contracts.py

```diff
@@ -0,0 +1,178 @@
+                (0, 0, {'name': 'Tuesday Afternoon', 'dayofweek': '1', 'hour_from': 13, 'hour_to': 16.6, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Wednesday Morning', 'dayofweek': '2', 'hour_from': 8, 'hour_to': 12, 'day_period': 'morning'})
+            ]
+        })
+
+        cls.employee_with_multiple_schedules = cls.env['hr.employee'].create({
+            'name': 'Employee with different working schedules',
+            'company_id': cls.belgian_company.id,
+        })
+        first_contract = cls.env['hr.contract'].create({
+            'name': "employee's contract",
+            'employee_id': cls.employee_with_multiple_schedules.id,
+            'resource_calendar_id': cls.resource_calendar.id,
+            'date_start': datetime.date(2024, 1, 1),
+            'date_end':   datetime.date(2024, 6, 30),
+            'wage': 2500
+        })
+        first_contract.write({'state': 'close'})
+        second_contract = first_contract.copy({
+            'resource_calendar_id': cls.resource_calendar_mid_time.id,
+            'date_start': datetime.date(2024, 7, 1),
+            'date_end': False
+        })
+        second_contract.write({'state': 'open'})
+
+    @users('test_user')
+    def test_accrual_plan_with_changing_working_schedules_1(self):
+        """
+        Accrual period 1 year, accrual value = 10 days
+        Employee worked from january to june in full time, then from july to december in half time.
+
+        Accrual plan is not based on worked time. Value for the year should be 7.5.
+        0,5 (accrual period) * 10 (accrual value) * 100% = 5 days
+        + 0,5 (accrual period) * 10 (accrual value) * 50% = 2.5 days
+        """
+
+        self.leave_type['company_id'] = self.belgian_company.id
+        allocation = self.env['hr.leave.allocation'].with_context(tracking_disable=True).create({
+            'name': 'Accrual allocation for employee',
+            'employee_id': self.employee_with_multiple_schedules.id,
+            'holiday_status_id': self.leave_type.id,
+            'number_of_days': 0,
+            'allocation_type': 'accrual',
+            'accrual_plan_id': self.yearly_accrual_plan.id,
+            'date_from': datetime.date(2024, 1, 1),
+        })
+        allocation.action_validate()
+
+        with freeze_time("2025-01-01"):
+            allocation._update_accrual()
+        self.assertAlmostEqual(allocation.number_of_days, 7.5, 1)
+
+    @users('test_user')
+    def test_accrual_plan_with_changing_working_schedules_2(self):
+        """
+        Accrual period 1 year, accrual value = 10 days
+        Employee worked from january to june in full time, then from july to december in half time.
+        Unpaid period in march, april, may.
+
+        Value should be computed like this :
+        0,5 (accrual period) * 10 (accrual value) * 100% (working rate) = 5 days (* based on worked time) = 2,5 days
```

Comment:
```suggestion
        0,5 (accrual period) * 10 (accrual value) * 100% (working rate) = 5 days *(3/6) = 2,5 days
```
for me it's easier to understand with the rate :D